pkg/plugins/models.go

```diff
@@ -280,6 +291,11 @@ func (e Error) AsErrorCode() ErrorCode {
 	return ""
 }
 
+func (e *Error) WithMessage(m string) *Error {
```

Comment:
I wanted to avoid that so this struct behaves as an `Error` as much as possible. If `Message` is exported, people may return `err.Message` rather than doing `err.Error()`, which behaves differently.

Reply:
I see. Then I think let's just keep this moving and see if we can/need to refactor later!